This adds a new field within the plat_stmmacenet_data structure, called has_gmac.
It also updates the stx7111_configure_ethernet according to the right Ethernet
configuration for the system config register 7.

Signed-off-by: Giuseppe Cavallaro <peppe.cavallaro@st.com>

Index: linux-2.6.23-stm/arch/sh/kernel/cpu/sh4/setup-stb7100.c
===================================================================
--- linux-2.6.23-stm.orig/arch/sh/kernel/cpu/sh4/setup-stb7100.c
+++ linux-2.6.23-stm/arch/sh/kernel/cpu/sh4/setup-stb7100.c
@@ -503,6 +503,7 @@ static void stb7109eth_hw_setup_null(voi
 static struct plat_stmmacenet_data eth7109_private_data = {
 	.bus_id = 0,
 	.pbl = 1,
+	.has_gmac = 0,
 	.fix_mac_speed = fix_mac_speed,
 	.hw_setup = stb7109eth_hw_setup_null,
 };
Index: linux-2.6.23-stm/arch/sh/kernel/cpu/sh4/setup-stx7200.c
===================================================================
--- linux-2.6.23-stm.orig/arch/sh/kernel/cpu/sh4/setup-stx7200.c
+++ linux-2.6.23-stm/arch/sh/kernel/cpu/sh4/setup-stx7200.c
@@ -547,10 +547,12 @@ static void fix_mac_speed(void *priv, un
 static struct plat_stmmacenet_data stmmaceth_private_data[2] = {
 {
 	.pbl = 32,
+	.has_gmac = 0,
 	.fix_mac_speed = fix_mac_speed,
 	.bsp_priv = (void*)0,
 }, {
 	.pbl = 32,
+	.has_gmac = 0,
 	.fix_mac_speed = fix_mac_speed,
 	.bsp_priv = (void*)1,
 } };
Index: linux-2.6.23-stm/include/linux/stm/soc.h
===================================================================
--- linux-2.6.23-stm.orig/include/linux/stm/soc.h
+++ linux-2.6.23-stm/include/linux/stm/soc.h
@@ -81,6 +81,7 @@ struct plat_lirc_data {
 struct plat_stmmacenet_data {
 	int bus_id;
 	int pbl;
+	int has_gmac;
 	void (*fix_mac_speed)(void *priv, unsigned int speed);
 	void (*hw_setup)(void);
 	void *bsp_priv;
